tests/transformers/test_transformers_model_export.py

```diff
@@ -3862,3 +3875,66 @@ def _calculate_expected_size(path_or_dir):
 
     mlmodel = yaml.safe_load(tmp_path.joinpath("MLmodel").read_bytes())
     assert mlmodel["model_size_bytes"] == expected_size
+
+
+@pytest.mark.parametrize("task", ["llm/v1/completions", "llm/v1/chat"])
+def test_text_generation_save_model_with_inference_task(task, text_generation_pipeline, model_path):
+    mlflow.transformers.save_model(
+        transformers_model=text_generation_pipeline,
+        path=model_path,
+        task=task,
+    )
+
+    mlmodel = yaml.safe_load(model_path.joinpath("MLmodel").read_bytes())
+
+    flavor_config = mlmodel["flavors"]["transformers"]
+    assert flavor_config["inference_task"] == task
+
+    assert mlmodel["metadata"]["task"] == task
+
+
+def test_text_generation_save_model_with_invalid_inference_task(
+    text_generation_pipeline, model_path
+):
+    with pytest.raises(
+        MlflowException, match=r"The task provided is invalid.*Must be.*llm/v1/completions"
+    ):
+        mlflow.transformers.save_model(
+            transformers_model=text_generation_pipeline,
+            path=model_path,
+            task="llm/v1/invalid",
+        )
+
+
+def test_text_generation_task_completions_predict_with_hf_params(
```

Comment:
Could we add a test for serving as well?

Reply:
Definitely, added `test_text_generation_task_completions_serve` now.
(As a future note, it still requires signature to be specified, and the request needs to be wrapped in `inputs` / `dataframe_split` etc, which would be addressed in followup PRs.)